drv/stm32h7-update-server/src/main.rs

```diff
@@ -279,6 +286,24 @@ impl<'a> ServerImpl<'a> {
 }
 
 impl idl::InOrderUpdateImpl for ServerImpl<'_> {
+    fn set_pending_boot_slot(
+        &mut self,
+        _: &RecvMessage,
+        slot: SlotId,
+    ) -> Result<(), RequestError<UpdateError>> {
+        if slot != self.pending {
+            self.swap_banks()?;
+        }
+        Ok(())
+    }
+
+    fn get_pending_boot_slot(
+        &mut self,
+        _: &RecvMessage,
+    ) -> Result<SlotId, RequestError<Infallible>> {
+        Ok(self.pending)
```

Comment:
Is this state  retrievable from a HW register? Just thinking of the corner case where update_server restarts and having two sources of truth.

Reply:
Ok... I see the fetch of HW state below. Looks ok.